pkg/resources/authentication_policy.go

```diff
@@ -0,0 +1,388 @@
+		Type: schema.TypeSet,
+		Elem: &schema.Schema{
+			Type:             schema.TypeString,
+			ValidateDiagFunc: StringInSlice([]string{"SAML", "PASSWORD"}, false),
+		},
+		Optional:    true,
+		Description: "A list of authentication methods that enforce multi-factor authentication (MFA) during login. Authentication methods not listed in this parameter do not prompt for multi-factor authentication. Allowed values are SAML and PASSWORD.",
+	},
+	"mfa_enrollment": {
+		Type:         schema.TypeString,
+		Optional:     true,
+		Description:  "Determines whether a user must enroll in multi-factor authentication. Allowed values are REQUIRED and OPTIONAL. When REQUIRED is specified, Enforces users to enroll in MFA. If this value is used, then the CLIENT_TYPES parameter must include SNOWFLAKE_UI, because Snowsight is the only place users can enroll in multi-factor authentication (MFA).",
+		ValidateFunc: validation.StringInSlice([]string{"REQUIRED", "OPTIONAL"}, false),
+		Default:      "OPTIONAL",
+	},
+	"client_types": {
+		Type: schema.TypeSet,
+		Elem: &schema.Schema{
+			Type:             schema.TypeString,
+			ValidateDiagFunc: StringInSlice([]string{"ALL", "SNOWFLAKE_UI", "DRIVERS", "SNOWSQL"}, false),
+		},
+		Optional:    true,
+		Description: "A list of clients that can authenticate with Snowflake. If a client tries to connect, and the client is not one of the valid CLIENT_TYPES, then the login attempt fails. Allowed values are ALL, SNOWFLAKE_UI, DRIVERS, SNOWSQL. The CLIENT_TYPES property of an authentication policy is a best effort method to block user logins based on specific clients. It should not be used as the sole control to establish a security boundary.",
+	},
+	"security_integrations": {
+		Type: schema.TypeSet,
+		Elem: &schema.Schema{
+			Type: schema.TypeString,
+		},
+		Optional:    true,
+		Description: "A list of security integrations the authentication policy is associated with. This parameter has no effect when SAML or OAUTH are not in the AUTHENTICATION_METHODS list. All values in the SECURITY_INTEGRATIONS list must be compatible with the values in the AUTHENTICATION_METHODS list. For example, if SECURITY_INTEGRATIONS contains a SAML security integration, and AUTHENTICATION_METHODS contains OAUTH, then you cannot create the authentication policy. To allow all security integrations use ALL as parameter.",
+	},
+	"comment": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Specifies a comment for the authentication policy.",
+	},
+	FullyQualifiedNameAttributeName: schemas.FullyQualifiedNameSchema,
+}
+
+// AuthenticationPolicy returns a pointer to the resource representing an authentication policy.
+func AuthenticationPolicy() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: CreateContextAuthenticationPolicy,
+		ReadContext:   ReadContextAuthenticationPolicy,
+		UpdateContext: UpdateContextAuthenticationPolicy,
+		DeleteContext: DeleteContextAuthenticationPolicy,
+
```

Comment:
Please add a description with a link to CREATE AUTHENTICATION POLICY - check row_access_policy.go

Reply:
done